spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -137,10 +142,15 @@ object Engine {
   @volatile private var _default: ThreadPool = null
 
   // Thread pool for layer use
-  @volatile private var _model: ThreadPool = new ThreadPool(1).setMKLThread(1)
+  @volatile private var _model: ThreadPool = new ThreadPool(1).setMKLThread(Env.mklNumThreads)
 
-  private def getCoreNumberFromEnv : Int = {
-    Option(System.getenv("DL_CORE_NUMBER")).map(_.toInt).getOrElse(getNumMachineCores)
+  /**
+   * return core number user defined.
+   */
+  private def getCoreNumberFromProperty() = {
+    val coreNumber = System.getProperty("bigdl.coreNumber", getNumMachineCores.toString).toInt
+    if (coreNumber > getNumMachineCores) getNumMachineCores
```

Comment:
if () {
} else {
}

BTW, I think we need to leave a comment here to explain why we do this. A log warning message may be better.

There's an assumption that user enable their HT, right? We may need to mention it in the comment.